upb/json/parser.rl

```diff
@@ -1325,27 +1457,103 @@ static void end_object(upb_json_parser *p) {
 
   value =
     number
-      >{ start_number(parser, p); }
-      %{ CHECK_RETURN_TOP(end_number(parser, p)); }
+      >{
+         if (is_top_level(parser)) {
```

Comment:
Could we move all of this logic to start_number() instead? I want to keep the Ragel machine part of the file small and simple.

Likewise for the other values below.